Week2/Section2_w2/MySpotify/Song.cpp

```diff
@@ -0,0 +1,74 @@
+#include "Song.h"
+#include <string>
+
+Song::Song()
+{
+	SongTitle = "";
+	ArtistName = "";
+	Duration = 0;
+}
+
+Song::Song(const std::string NewSongTitle, const std::string NewArtistName, const int NewDuration)
+{
+	SongTitle = NewSongTitle;
+	ArtistName = NewArtistName;
+	Duration = NewDuration;
+}
+
+std::string Song::GetFormatedDuration()
+{
+	std::string Minutes = std::to_string(Duration / 100);
+	std::string Seconds = std::to_string(Duration - (Duration / 100) * 100);
+	return Minutes + ":" + Seconds;
+}
+
+bool Song::IsValid()
+{
+	if (isupper(SongTitle[0]))
+	{
+		return true; //if the song name starts with an uppercase letter, is valid
```

Comment:
Oddly specific but nice 👀 However, what about an empty artist name? Or a negative duration?

Reply:
Also, keep in mind that this could cause issues if the SongTitle was empty, since you're trying to access the first character.

Reply:
okok I'll add that conditions